src/main/java/uk/ac/cam/cl/dtg/segue/dao/content/LLMMarkingExpressionDeserializer.java

```diff
@@ -0,0 +1,48 @@
+package uk.ac.cam.cl.dtg.segue.dao.content;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import uk.ac.cam.cl.dtg.isaac.dos.content.LLMMarkingConstant;
+import uk.ac.cam.cl.dtg.isaac.dos.content.LLMMarkingExpression;
+import uk.ac.cam.cl.dtg.isaac.dos.content.LLMMarkingFunction;
+import uk.ac.cam.cl.dtg.isaac.dos.content.LLMMarkingVariable;
+
+import java.io.IOException;
+
+public class LLMMarkingExpressionDeserializer extends JsonDeserializer<LLMMarkingExpression> {
+    private final ObjectMapper objectMapper;
+
+    public LLMMarkingExpressionDeserializer(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public LLMMarkingExpression deserialize(final JsonParser jsonParser, final DeserializationContext deserializationContext)
+            throws IOException {
+        ObjectNode root = objectMapper.readTree(jsonParser);
```

Comment:
I.e. why not just do `
```suggestion
        ObjectMapper mapper = (ObjectMapper) jsonParser.getCodec();
        ObjectNode root = objectMapper.readTree(jsonParser);
```
like all the other deserialisers do?